libp2p-websocket/src/desktop.rs

```diff
@@ -191,6 +191,24 @@ where
 
 		Ok(Box::new(dial) as Box<_>)
 	}
+
+    fn nat_traversal(&self, server: &Multiaddr, observed: &Multiaddr) -> Option<Multiaddr> {
```

Comment:
it seems like, `nat_traversal` always consume `Multiaddr`. maybe the signature of the function should move the values instead of taking the reference?

Reply:
The implementations are supposed to either return `None` or build a `Multiaddress` that is different from `server` and `observed`.

It could indeed be possible to take ownership of one of the multiaddresses and modify it (instead of building a brand new one). However the API of rust-multiaddr doesn't really allow modifying a multiaddress in place, because a multiaddr is stored in memory as a raw array of bytes.

Since doing this change is difficult, I'd suggest merging this PR and opening an issue instead?

Reply:
fait enough 👍